sdk/attestation/Azure.Security.Attestation/src/Models/AttestationTokenSigningKey.cs

```diff
@@ -45,7 +45,11 @@ public AttestationTokenSigningKey(X509Certificate2 certificate)
             {
                 throw new ArgumentException($"Certificate provided {certificate.ToString()} does not have a private key.");
             }
+#if NET6_0_OR_GREATER
+            Signer = (AsymmetricAlgorithm)certificate.GetRSAPublicKey() ?? certificate.GetDSAPrivateKey();
+#else
             Signer = certificate.PrivateKey;
```

Comment:
See: [SYSLIB0028: X509Certificate2.PrivateKey is obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0028)